### PR TITLE
Match latest policy patterns for Arc

### PR DESCRIPTION
**Pull request by mgreenegit**

### CODE REVIEW

File: samples/Automation/onboard-to-automation-dsc/azurepolicy.json [hunk 1]
```diff
@@ -181,140 +185,154 @@
...
+    "automationAccount": {
+      "type": "String",
+      "metadata": {
+        "displayName": "Automation account",
+        "description": "Select Automation account from dropdown list. If this account is outside of the scope of the assignment you must manually grant 'Contributor' permissions (or similar) on the Automation account to the policy assignment's principal ID.",
+        "strongType": "Microsoft.Automation/automationAccounts",
+        "assignPermissions": true
       }
+    },
+    "IncludeArcMachines": {
```
Comment (pilor) [inline samples/Automation/onboard-to-automation-dsc/azurepolicy.json:325]: >IncludeArcMachines [](start = 5, length = 18)

The other files in this folder need to be updated

File: samples/Compute/deploy-dsc-extension/azurepolicy.json [hunk 2]
```diff
@@ -248,8 +228,8 @@
                   {
                     "condition": "[equals(toLower(parameters('type')), toLower('Microsoft.Compute/virtualMachines'))]",
                     "type": "Microsoft.Compute/virtualMachines/extensions",
-                    "name": "[concat(parameters('machineName'),'/DSC')]",
-                    "apiVersion": "2019-07-01",
+                    "name": "[concat(parameters('machineName'),'/Microsoft.Powershell.DSC')]",
+                    "apiVersion": "2020-06-01",
```
Comment (pilor) [inline samples/Compute/deploy-dsc-extension/azurepolicy.json:232]: >apiVersion [](start = 21, length = 10)

update azurepolicy.rules.json as well

**Comment by mgreenegit** [comment]
@pilor question - is there a preferred way to handle the following problem?

Hybrid VM extension does not yet support the same locations as Compute extension. As a result, even though there is a condition on the Hybrid extension, ARM is evaluating the resource and returning an error if the Compute VM is in a location where Hybrid does not allow.

```
The provided location 'northcentralus' is not available for resource type 'Microsoft.HybridCompute/machines/extensions'. List of available regions for the resource type is 'westeurope,westus2,southeastasia,eastus'.
```

**Comment by pilor** [comment]
> @pilor question - is there a preferred way to handle the following problem?
> 
> Hybrid VM extension does not yet support the same locations as Compute extension. As a result, even though there is a condition on the Hybrid extension, ARM is evaluating the resource and returning an error if the Compute VM is in a location where Hybrid does not allow.
> 
> ```
> The provided location 'northcentralus' is not available for resource type 'Microsoft.HybridCompute/machines/extensions'. List of available regions for the resource type is 'westeurope,westus2,southeastasia,eastus'.
> ```
> 
> I can make "Arc location" a parameter, but there is no way to make it contingent upon opt-in for Arc.
> 
> Another option would be to split this across two definitions.

You'd have to ask the deployment team.  That type of validation should not happen for a resource that isn't going to be deployed due to a false condition imo.

File: samples/Automation/onboard-to-automation-dsc/azurepolicy.json [hunk 3]
```diff
@@ -225,20 +273,25 @@
                   }
                 },
                 "variables": {
-                  "automationAccountName": "[last(split(parameters('automationAccount'),'/'))]"
+                  "automationAccountName": "[last(split(parameters('automationAccount'),'/'))]",
+                  "location": {
```
Comment (mgreenegit) [inline samples/Automation/onboard-to-automation-dsc/azurepolicy.json:277]: @bmoore-msft this workaround does handle the issue I was fighting around template validation. What is your opinion of this?

**Comment by mgreenegit** [comment]
@pilor I will submit this for consideration as a built-in supported by the Automation team. Will close this PR after Brian takes a look.

**Pull request closed without merging**